allennlp/models/crf_tagger.py

```diff
@@ -81,20 +104,45 @@ def __init__(self, vocab: Vocabulary,
                                                            self.num_tags))
 
         if constraint_type is not None:
+            warnings.warn("'constraint_type' was removed and replaced with"
+                          "'label_encoding', 'constrain_crf_decoding', and "
+                          "'calculate_span_f1' in version 0.6.1. It will be "
+                          "removed in version 0.8.", DeprecationWarning)
+            label_encoding = constraint_type
+            constrain_crf_decoding = True
+        self.label_encoding = label_encoding
+        if constrain_crf_decoding:
+            if not label_encoding:
+                raise ConfigurationError("constrain_crf_decoding is True, but "
+                                         "no label_encoding was specified.")
             labels = self.vocab.get_index_to_token_vocabulary(label_namespace)
-            constraints = allowed_transitions(constraint_type, labels)
+            constraints = allowed_transitions(label_encoding, labels)
         else:
             constraints = None
 
+        self.include_start_end_transitions = include_start_end_transitions
         self.crf = ConditionalRandomField(
                 self.num_tags, constraints,
                 include_start_end_transitions=include_start_end_transitions
         )
 
-        self.span_metric = SpanBasedF1Measure(vocab,
-                                              tag_namespace=label_namespace,
-                                              label_encoding=constraint_type or "BIO")
-
+        self.metrics = {
+                "accuracy": CategoricalAccuracy(),
+                "accuracy3": CategoricalAccuracy(top_k=3)
+        }
+        self.calculate_span_f1 = calculate_span_f1
+        if calculate_span_f1:
+            if not label_encoding:
+                raise ConfigurationError("calculate_span_f1 is True, but "
+                                         "no label_encoding was specified.")
+            self._f1_metric = SpanBasedF1Measure(vocab,
+                                                 tag_namespace=label_namespace,
+                                                 label_encoding=label_encoding)
+        elif constraint_type is not None:
+            # Maintain deprecated behavior if constraint_type is provided
+            self._f1_metric = SpanBasedF1Measure(vocab,
+                                                 tag_namespace=label_namespace,
+                                                 label_encoding=constraint_type or "BIO")
```

Comment:
if you're in this case, doesn't it have to be true that `constraint_type` was provided?, so you don't need the `or "BIO"`?

Reply:
you're right, thanks.